src/pages/SearchPage.js

```diff
@@ -1,37 +1,38 @@
 import React from 'react';
-import movies from '../mocks/movie';
+import {connect} from 'react-redux';
+
+import {startMoviesRequest} from '../actions/movies';
+import moviesSelector from '../selectors/movies';
 import SearchHeader from '../components/SearchHeader';
 import ResultHeader from '../components/ResultHeader';
 import MovieList from '../components/MovieList';
 import Footer from '../components/Footer';
 
-export default class SearchPage extends React.PureComponent {
-
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      sortBy: 'release_date'
-    }
-  }
-
-  onSortChange = (sortBy) => {
-    this.setState({sortBy: sortBy});
-  }
-
+export class SearchPage extends React.PureComponent {
   render() {
+    const {movies} = this.props;
     // Uncomment this line to test ErrorBoundary component
     // throw new Error('Something went wrong');
     return (
       <div className="search-page">
         <SearchHeader />
-        <ResultHeader resultNumber={movies.length}
-                      sortBy={this.state.sortBy}
-                      onChange={this.onSortChange} />
-        <MovieList movies={movies} 
-                   onPageChange={this.props.onPageChange}/>
+        {!!movies.length && (
+          <React.Fragment>
+          <ResultHeader resultNumber={movies.length} />
+          <MovieList movies={movies} 
```

Comment:
As `MovieList` is an specialized component to handle the movie list, you could directly wire it to redux instead of passing this props.

Reply:
The problem here is that I use MovieList for two different lists, the search list and the suggestions list, and they are completely different. How can I know from the component what part of the state of redux to use?